File: read_excel.js

const XLSX = require('xlsx');
const path = require('path');

const file = path.join(__dirname, 'Cash_Tracker_With_Warning.xlsx');
try {
    const workbook = XLSX.readFile(file);
    console.log('Sheet names:', workbook.SheetNames);
    
    workbook.SheetNames.forEach(sheetName => {
        console.log('\n=== Sheet:', sheetName, '===');
        const sheet = workbook.Sheets[sheetName];
        const data = XLSX.utils.sheet_to_json(sheet, { header: 1 });
        console.log('Data:', data.slice(0, 10)); // First 10 rows
    });
} catch (e) {
    console.error('Error reading Excel file:', e);
}

File: read_excel.py

import pandas as pd

try:
    excel_file = 'Cash_Tracker_With_Warning.xlsx'
    
    # Read all sheets
    xl = pd.ExcelFile(excel_file)
    print("Sheet names in Excel file:")
    for sheet_name in xl.sheet_names:
        print(f"- {sheet_name}")
        print()
        df = pd.read_excel(excel_file, sheet_name=sheet_name)
        print("First 10 rows:")
        print(df.head(10))
        print()
        print("Columns:")
        print(df.columns.tolist())
        print()
        print("Data types:")
        print(df.dtypes)
        print()
        print("-" * 80)
except Exception as e:
    print(f"Error reading Excel file: {e}")
